src/Android/Xamarin.Android/Samples/Analysis/LineOfSightGeoElement/LineOfSightGeoElement.cs

```diff
@@ -0,0 +1,286 @@
+            // 4. Add the observer to the overlay
+            overlay.Graphics.Add(observerGraphic);
+            // 5. Add the overlay to the scene
+            _mySceneView.GraphicsOverlays.Add(overlay);
+
+            // Add the taxi to the scene
+            // 1. Create the model symbol for the taxi
+            ModelSceneSymbol taxiSymbol = await ModelSceneSymbol.CreateAsync(await GetModelUri());
+            // 2. Set the anchor position for the mode; ensures that the model appears above the ground
+            taxiSymbol.AnchorPosition = SceneSymbolAnchorPosition.Bottom;
+            // 3. Create the graphic from the taxi starting point and the symbol
+            _taxiGraphic = new Graphic(points[0], taxiSymbol);
+            // 4. Add the taxi graphic to the overlay
+            overlay.Graphics.Add(_taxiGraphic);
+
+            // Create GeoElement Line of sight analysis (taxi to building)
+            // 1. Create the analysis
+            _geoLine = new GeoElementLineOfSight(observerGraphic, _taxiGraphic);
+            // 2. Apply an offset to the target. This helps avoid some false negatives
+            _geoLine.TargetOffsetZ = 2;
+            // 3. Create the analysis overlay
+            AnalysisOverlay myAnalysisOverlay = new AnalysisOverlay();
+            // 4. Add the analysis to the overlay
+            myAnalysisOverlay.Analyses.Add(_geoLine);
+            // 5. Add the analysis overlay to the scene
+            _mySceneView.AnalysisOverlays.Add(myAnalysisOverlay);
+
+            // 1. Create a timer; this will enable animating the taxi
+            var timer = new Timer(60);
+            // 2. Move the taxi every time the timer expires
+            timer.Elapsed += AnimationTimer_Elapsed;
+            // 3. Keep the timer running continuously
+            timer.AutoReset = true;
+            // 4. Start the timer
+            timer.Start();
+
+            // Subscribe to TargetVisible events; allows for updating the UI and selecting the taxi when it is visible
+            _geoLine.TargetVisibilityChanged += Geoline_TargetVisibilityChanged;
+
+            // Add the scene to the view
+            _mySceneView.Scene = myScene;
+        }
+
+        private void AnimationTimer_Elapsed(object sender, EventArgs e)
+        {
+            // Note: the contents of this function are solely related to animating the taxi
+
+            // Increment the frame counter
+            frameIndex++;
+
+            // Reset the frame counter once one segment of the path has been travelled
+            frameIndex = frameIndex % frameMax;
+            if (frameIndex == 0)
+            {
+                // Start navigating toward the next point; restart circuit if finished
+                pointIndex = (pointIndex + 1) % points.Length;
+            }
+
+            // Get the point the taxi is travelling from
+            MapPoint starting = points[pointIndex % points.Length];
```

Comment:
This is done to keep it consistent with the line below. `points[(pointIndex + 1) % points.Length` uses `% points.Length` to keep the index within the bounds of the array, rather than needing to use a conditional and manually reset the pointIndex. It is true that pointIndex itself will never go out of bounds as written, but you'd need to understand the rest of the code to know that. The idea was that the extra check would make the pattern more explicit.